tools/functional-tester/etcd-runner/main.go

```diff
@@ -212,6 +219,166 @@ func runRacer(eps []string, round int) {
 	doRounds(rcs, round)
 }
 
+func runWatcher(eps []string) {
+	var runningTime time.Duration = 30 // time for which operation should be performed
+	noOfPrefixes := 36                 // total number of prefixes which will be watched upon
+	watchPerPrefix := 10               // number of watchers per prefix
+	reqRate := 20                      // put request per second
+	keyPrePrefix := 100000             // max number of keyPrePrefixs for put operation
+
+	ctx := context.Background()
+	prefixKeysMap := make(map[string][]string) // map for holding prefixes and values
+	// pre generate prefixes equal to noOfPrefixes
+	prefixes := generateUniqueKeys(5, noOfPrefixes)
+	// pre generate keys
+	keys := generateRandomKeys(10, keyPrePrefix)
+
+	var (
+		revision int64
+		wg       sync.WaitGroup
+		pr       *clientv3.PutResponse
+		gr       *clientv3.GetResponse
+		err      error
+	)
+
+	for i := range prefixes {
+		prefix := prefixes[i]
+		// initialize prefix key map for each prefix
+		prefixKeysMap[prefix] = make([]string, 0)
```

Comment:
@xiang90 Please find below my comments for the suggestions:
 ````// initialize the map with all prefixes
    for _, prefix := range prefixes {
        prefixKeysMap[prefix] = make([]string, 0)
    }
``` 
This is required since in the second section where we are checking and managing all keys we need all prefixes to be already present in the map, else the behavior will not work
2.```remove the default case ``` 
Will definatly remove the default section.
3. ```exit when the number of received key matches what you expect.```
Will add it.